openedx/core/lib/api/test_utils.py

```diff
@@ -7,14 +7,14 @@
 import re
 import six
 
-from django.test import TestCase
+from django.test import TransactionTestCase
 from django.test.utils import override_settings
 
 TEST_API_KEY = "test_api_key"
 
 
 @override_settings(EDX_API_KEY=TEST_API_KEY)
-class ApiTestCase(TestCase):
+class ApiTestCase(TransactionTestCase):
```

Comment:
What forced this change?

Reply:
6 tests in https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/views/tests/test_register.py failed with following error
`django.db.transaction.TransactionManagementError: An error occurred in the current transaction. You can't execute queries until the end of the 'atomic' block.` I think the same queries are causing this for which we had to update the query counts in files up.

Reply:
Often that's an indication of another problem that needs to be fixed.  Could you try those tests again without this change?  To make sure you didn't already fix the problem with one of the other changes, and to get more detailed error information if it does still happen.  Part of my concern with the change is that a lot of tests use this as a base class, and usually TestCase is a better choice.

Reply:
Hey @jmbowman I reverted the change and 6 tests failed again, you can see the logs here at https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/21108/

Reply:
That did turn up some interesting problems; it looks like it gets caught in a loop where it hits an error, tries to render a 500 page but that hits another error, so it tries to render the 500 page again but hits the same error again, etc.  I think we need to fix some of the error handling so the 500 page successfully renders and tells us what the original exception was.  At least one of the failures in that job was from trying to get the value of a waffle switch when rendering a 500 page after the transaction is already invalidated.  I tried to reproduce locally, but instead hit a similar issue with trying to load the site configuration object.

I'm inclined to put this on the back burner until we get the celery and Python 3.8 upgrades out.  After that, we can try to fix the 500 page rendering so it doesn't hit this exception loop of doom.

Reply:
Sure. Let's delay this until we are done with python 3.8.